src/python/pants/backend/python/typecheck/mypy/rules_integration_test.py

```diff
@@ -347,6 +350,59 @@ def add(x: int, y: int) -> str:
     assert f"{PACKAGE}/math/add.py:5" in result[0].stdout
 
 
+@skip_unless_python27_present
+def test_works_with_python27(rule_runner: RuleRunner) -> None:
+    """A regression test that we can properly handle Python 2-only third-party dependencies.
+
+    There was a bug that this would cause the runner PEX to fail to execute because it did not have
+    Python 3 distributions of the requirements.
+
+    This support is not as robust as we'd like. We'll only use third-party distributions if its
+    wheel is also compatible with the Python 3 interpreter being used to run MyPy.
+    https://github.com/pantsbuild/pants/issues/10819 tracks improving this, which might not even be
+    possible.
+    """
+    rule_runner.add_to_build_file(
+        "",
+        dedent(
+            """\
+            # Both requirements are a) typed and b) compatible with Py2 and Py3. However, `x690`
+            # has a distinct wheel for Py2 vs. Py3, whereas libumi has a universal wheel. We only
+            # expect libumi to be usable by MyPy.
+
+            python_requirement_library(
+                name="libumi",
+                requirements=["libumi==0.0.2"],
+            )
+
+            python_requirement_library(
+                name="x690",
+                requirements=["x690==0.2.0"],
+            )
+            """
+        ),
+    )
+    source_file = FileContent(
+        f"{PACKAGE}/py2.py",
+        dedent(
+            """\
+            from libumi import hello_world
+            from x690 import types
+
+            print "Blast from the past!"
+            print hello_world() - 21  # This should fail. You can't subtract an `int` from `bytes`.
```

Comment:
This is a mypy failure, right? We don't execute the code in this?

Reply:
Yes, exactly. Will clarify.